src/Language/Futhark/TypeChecker/Terms.hs

```diff
@@ -536,7 +536,12 @@ checkExp (Range start maybe_step end loc) = do
   let start_t = toStructural $ typeOf start'
   maybe_step' <- case maybe_step of
     Nothing -> return Nothing
-    Just step -> Just <$> (require [start_t] =<< checkExp step)
+    Just step -> do
+      case (start, step) of
+        ((Literal (SignedValue (Int32Value x)) _), (Literal (SignedValue (Int32Value y)) _)) -> do
+          when (x == y) $ warn loc "First and second element of range are identical, this will produce an empty array."
```

Comment:
Since `start` and `step` don't conform to `Eq`, should I warn when `(Literal x _, Literal y _) && x == y` and also `(Var x_name _ _, Var y_name _ _) where x_name == y_name`?